### PR TITLE
Introduced support for ARM64 Mariner 2.0

### PR DESCRIPTION
**Pull request by SeanDougherty**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
in support of ADO deliverable #40097182

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: packer.mk [hunk 1]
```diff
@@ -37,6 +37,21 @@ else
 	@packer build -var-file=vhdbuilder/packer/settings.json vhdbuilder/packer/vhd-image-builder-mariner.json
 endif
 else ifeq (${OS_VERSION},V2)
+ifeq (${ARCHITECTURE}, ARM64)
+ifeq (${MODE},gen2Mode)
+	@echo "${MODE}: Building with Hyper-v generation 2 ARM64 VM"
+	@rm -f ./vhdbuilder/packer/packer
+	@curl -fsSL https://packerbinary.blob.core.windows.net/bin/packer -o ./vhdbuilder/packer/packer
```
Comment (chengliangli0918) [inline packer.mk:44]: We don't need to use my private packer builds to workaround an ARM64 os disk capturing issue. Please see my recent PR https://github.com/Azure/AgentBaker/pull/1956 to make corresponding changes

File: vhdbuilder/packer/vhd-image-builder-mariner2-arm64-gen2.json [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+          "now": "{{user `create_time`}}",
+          "createdBy": "aks-vhd-pipeline"
+        },
+        "location": "{{user `location`}}",
+        "vm_size": "{{user `vm_size`}}",
+        "custom_data_file": "vhdbuilder/packer/sha1rsa_custom_data",
+        "managed_image_os_disk_snapshot_name": "{{user `arm64_os_disk_snapshot_name`}}",
+        "managed_image_resource_group_name": "{{user `resource_group_name`}}",
+        "managed_image_name": "{{user `sig_image_name`}}-{{user `gen2_captured_sig_version`}}",
+        "shared_image_gallery_destination": {
```
Comment (chengliangli0918) [inline vhdbuilder/packer/vhd-image-builder-mariner2-arm64-gen2.json:49]: Please refer to my recent AB PR https://github.com/Azure/AgentBaker/pull/1956 for corresponding update to this file.

File: .pipelines/.vsts-vhd-builder-release.yaml [hunk 3]
```diff
@@ -352,6 +356,31 @@ stages:
...
+    - job: build
+      timeoutInMinutes: 180
+      steps:
+      - bash: |
+          echo '##vso[task.setvariable variable=DRY_RUN]${{parameters.dryrun}}'
+          echo '##vso[task.setvariable variable=OS_SKU]CBLMariner'
+          echo '##vso[task.setvariable variable=OS_VERSION]V2'
+          echo '##vso[task.setvariable variable=IMG_PUBLISHER]MicrosoftCBLMariner'
+          echo '##vso[task.setvariable variable=IMG_OFFER]cbl-mariner'
+          echo '##vso[task.setvariable variable=IMG_SKU]cbl-mariner-2-gen2-arm64'
```
Comment (chengliangli0918) [inline .pipelines/.vsts-vhd-builder-release.yaml:372]: The SKU of arm64 mariner 2.0 doesn't have '-gen2'. I double checked with Chris Co before, here is his answer
"Since Azure only supports ARM64 GEN2 VMs, our ARM64 Marketplace image (cbl-mariner-2-arm64) is actually a Gen2 image"


image

File: vhdbuilder/packer/vhd-image-builder-mariner2-arm64-gen2.json [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+          "resource_group": "{{user `resource_group_name`}}",
+          "gallery_name": "PackerSigGalleryWestUS2",
```
Comment (chengliangli0918) [inline vhdbuilder/packer/vhd-image-builder-mariner2-arm64-gen2.json:51]: I think we don't need to create a separate galleryname containing 'westus2'. We have to run arm64 vhd build now in 'westus2' because Arm64 Compute is under Public Preview, and there is no ARM64 resource in EastUs. Once ARM64 Compute is in GA and there is ARM64 resource in EastUS. We'll build ARM64 VHD in EastUS region just as all AMD64 VHDs by simply changing AZURE_LOCATION in .vsts-vhd-builder-release.yaml. So if you use 'westus2' in resource names, we have to update in a few other places.

File: packer.mk (hunk 1 above)
Comment (SeanDougherty) [reply packer.mk]: Addressed with a git commit --amend, thank you!

File: vhdbuilder/packer/vhd-image-builder-mariner2-arm64-gen2.json (hunk 2 above)
Comment (SeanDougherty) [reply vhdbuilder/packer/vhd-image-builder-mariner2-arm64-gen2.json]: Similarly addressed!

File: .pipelines/.vsts-vhd-builder-release.yaml (hunk 3 above)
Comment (SeanDougherty) [reply .pipelines/.vsts-vhd-builder-release.yaml]: Thank you for catching this!

File: vhdbuilder/packer/vhd-image-builder-mariner2-arm64-gen2.json (hunk 4 above)
Comment (SeanDougherty) [reply vhdbuilder/packer/vhd-image-builder-mariner2-arm64-gen2.json]: Gotcha, that makes sense

File: .pipelines/.vsts-vhd-builder-release.yaml [hunk 5]
```diff
@@ -53,6 +53,10 @@ parameters:
   displayName: Build MarinerV2 Gen2
   type: boolean
   default: true
+- name: buildMarinerV2gen2ARM64
```
Comment (chengliangli0918) [inline .pipelines/.vsts-vhd-builder-release.yaml:56]: Could you ask @alexeldeib to grant you write permission on AgentBaker repo, so that you can make your change on your branch on AgentBaker repo instead of your fork. Then you can test the Mariner 2.0 ARM64 build in pipeline https://msazure.visualstudio.com/CloudNativeCompute/_build?definitionId=172842

Comment (SeanDougherty) [reply .pipelines/.vsts-vhd-builder-release.yaml:56]: Pipeline has been [executed ](https://msazure.visualstudio.com/CloudNativeCompute/_build/results?buildId=58216239&view=results). There were two images which failed to build. These were both related to insufficient cores in the standardNCSv3Family Cores quota.

Additionally, [this pipeline](https://msazure.visualstudio.com/CloudNativeCompute/_build/results?buildId=58214924&view=results) was automatically triggered when my branch was pushed to AB's repository. It appears to have succeeded, no issue.

**Review by chengliangli0918 — APPROVED** [review]

File: .pipelines/.vsts-vhd-builder-release.yaml (hunk 5 above)
Comment (chengliangli0918) [reply .pipelines/.vsts-vhd-builder-release.yaml:56]: Two Gen2+GPU failures because of standardNCSv3Family Cores quota are expected. I approved your PR. Not sure if you can merge yourself. If you cannot, you can ask Henry in your team or me to merge on behalf.

File: vhdbuilder/packer/convert-osdisk-snapshot-to-sig.sh [hunk 6]
```diff
@@ -21,6 +21,6 @@ echo "ManagedImageSharedImageGalleryId: /subscriptions/${SUBSCRIPTION_ID}/resour
 
 disk_snapshot_id="/subscriptions/${SUBSCRIPTION_ID}/resourceGroups/${AZURE_RESOURCE_GROUP_NAME}/providers/Microsoft.Compute/snapshots/${ARM64_OS_DISK_SNAPSHOT_NAME}"
 az snapshot update --resource-group ${AZURE_RESOURCE_GROUP_NAME} -n ${ARM64_OS_DISK_SNAPSHOT_NAME} --architecture Arm64
-az sig image-version create --location $AZURE_LOCATION --resource-group ${AZURE_RESOURCE_GROUP_NAME} --gallery-name PackerSigGalleryEastUS \
+az sig image-version create --location $AZURE_LOCATION --resource-group ${AZURE_RESOURCE_GROUP_NAME} --gallery-name ${SIG_GALLERY_NAME} \
```
Comment (chengliangli0918) [inline vhdbuilder/packer/convert-osdisk-snapshot-to-sig.sh:24]: Please revert this change, convert-osdisk-snapshot-to-sig.sh doesn't have env variable ${SIG_GALLERY_NAME}.  Ubuntu 1804 arm64 build fail (https://msazure.visualstudio.com/CloudNativeCompute/_build/results?buildId=58231230&view=logs&j=b03669c0-70b9-5993-d66f-d20387ecabc8&s=13a1e9be-10b9-5809-47f7-c38e14a56b67)

**Review by chengliangli0918 — CHANGES_REQUESTED** [review]
I saw failure on arm64 ubuntu 18.04 build. https://msazure.visualstudio.com/CloudNativeCompute/_build/results?buildId=58231230&view=logs&j=b03669c0-70b9-5993-d66f-d20387ecabc8&s=13a1e9be-10b9-5809-47f7-c38e14a56b67

**Review by chengliangli0918 — APPROVED** [review]
